File: sdkwork-backend-sdk-flutter/lib/src/api/rbac_role.dart
import '../http/client.dart';
import '../models.dart';

class RbacRoleApi {
  final HttpClient _client;
  
  RbacRoleApi(this._client);

  /// Update an existing system role
  Future<PlusApiResultPlusRoleVO?> update(PlusRoleForm body) async {
    final response = await _client.put(ApiPaths.backendPath('/role'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusRoleVO ? response : null;
  }

  /// Create a new system role
  Future<PlusApiResultPlusRoleVO?> create(PlusRoleForm body) async {
    final response = await _client.post(ApiPaths.backendPath('/role'), body: body, contentType: 'application/json');
    return response is PlusApiResultPlusRoleVO ? response : null;
  }

  /// Get system roles by page
  Future<PlusApiResultPagePlusRoleVO?> listByPage(QueryListForm? body, Map<String, dynamic>? params) async {
    final response = await _client.post(ApiPaths.backendPath('/role/list'), body: body, params: params, contentType: 'application/json');
    return response is PlusApiResultPagePlusRoleVO ? response : null;
  }

  /// Get all system roles
  Future<PlusApiResultListPlusRoleVO?> listAllEntities(QueryListForm? body) async {
    final response = await _client.post(ApiPaths.backendPath('/role/list/all'), body: body, contentType: 'application/json');
    return response is PlusApiResultListPlusRoleVO ? response : null;
  }

  /// Get a system role by ID
  Future<PlusApiResultPlusRoleVO?> getById(String id) async {
    final response = await _client.get(ApiPaths.backendPath('/role/${id}'));
    return response is PlusApiResultPlusRoleVO ? response : null;
  }

  /// Delete a system role
  Future<PlusApiResultBoolean?> delete(String id) async {
    final response = await _client.delete(ApiPaths.backendPath('/role/${id}'));
    return response is PlusApiResultBoolean ? response : null;
  }
}

File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/api/rbac_role.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultBoolean, PlusApiResultListPlusRoleVO, PlusApiResultPagePlusRoleVO, PlusApiResultPlusRoleVO, PlusRoleForm, QueryListForm

class RbacRoleApi:
    """rbac_role API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def update(self, body: PlusRoleForm) -> PlusApiResultPlusRoleVO:
        """Update an existing system role"""
        return self._client.put(f"/backend/v3/api/role", json=body)

    def create(self, body: PlusRoleForm) -> PlusApiResultPlusRoleVO:
        """Create a new system role"""
        return self._client.post(f"/backend/v3/api/role", json=body)

    def list_by_page(self, body: Optional[QueryListForm] = None, params: Optional[Dict[str, Any]] = None) -> PlusApiResultPagePlusRoleVO:
        """Get system roles by page"""
        return self._client.post(f"/backend/v3/api/role/list", json=body, params=params)

    def list_all_entities(self, body: Optional[QueryListForm] = None) -> PlusApiResultListPlusRoleVO:
        """Get all system roles"""
        return self._client.post(f"/backend/v3/api/role/list/all", json=body)

    def get_by_id(self, id: str) -> PlusApiResultPlusRoleVO:
        """Get a system role by ID"""
        return self._client.get(f"/backend/v3/api/role/{id}")

    def delete(self, id: str) -> PlusApiResultBoolean:
        """Delete a system role"""
        return self._client.delete(f"/backend/v3/api/role/{id}")
